server/container_create_linux.go

```diff
@@ -1119,6 +1136,35 @@ func addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container, mountLabel,
 			options = append(options, "rprivate")
 		}
 
+		var mountOption string
+		if m.Readonly {
+			mountOption = "ro"
+			if m.RecursiveReadOnly {
+				if !rroSupport {
+					return nil, nil, fmt.Errorf(
+						"recursive read-only mount is not supported for hostPath %q",
+						m.HostPath,
+					)
+				}
+				if m.Propagation != types.MountPropagation_PROPAGATION_PRIVATE {
+					return nil, nil, fmt.Errorf(
+						"recursive read-only mount requires private propagation for hostPath %q, got: %s",
+						m.HostPath, m.Propagation,
+					)
+				}
+				mountOption = "rro"
+			}
+		} else {
+			if m.RecursiveReadOnly {
+				return nil, nil, fmt.Errorf(
+					"recursive read-only mount conflicts with read-write mount for hostPath %q",
+					m.HostPath,
+				)
+			}
+			mountOption = "rw"
+		}
+		options = append(options, mountOption)
```

Comment:
we can do this with a flat if/else:
```
if rro && ro {
// rro
} else if rro
// err
} else if ro {
// ro
} else {
// rw
}
```
which I think would be easier to read

Reply:
Sure. I can do it like that.

For the record, I personally find nested if-else-if blocks harder to read.

Reply:
yeah It would be best as a switch but we have two variables to check :(

Reply:
Done.

Reply:
I am going to rewrite this using a switch statement, since linter picks up on it, per:

```golang
server/container_create_linux.go:1119:3: ifElseChain: rewrite if-else to switch statement (gocritic)
		if m.Readonly && m.RecursiveReadOnly {
```